### DISCUSSION TITLE
Binary numbers.

### DISCUSSION THREAD

**Discussion by VBFNiya** [twinbasic/twinbasic#890] [Ideas]
It's been a while since I been here. Hope everyone is doing fine. This is a suggestion I wanted to make for a long time but I kept forgetting. I just remembered it again and decided to say something before I forget.

This is something I've long wanted in BASIC for as long as I could remember and someone else may have suggested it already but just in case it hasn't been, I will mention it. It's a feature that I've seen in assemblers, the ability to define numbers in binary. 

For example in VB6 and VB.Net we could define a number in hex using this:-
```vb6
Dim a As Byte = &HFF 'a = 255
```
I would kill to be able to do this:-
```vb6
Dim a As Byte = &B00001000 ' a = 8
```
Currently we have to do it like this in VB.Net:-
```vb.net
Dim a As Byte = Convert.ToInt32("00001000", 2)
```
It's not a first class citizen in VB.Net and god only knows how annoying it would be to implement in VB6.

I think the BASIC language would be that much richer with this feature. 

EDIT:

Ah It seems VB.Net already has it I was just never aware. Nonetheless point still stands.

**Comment by bclothier** [comment]
This already exists in twinBASIC.

```
    Public Sub BinaryIt()
        Debug.Print &B001
        Debug.Print &B010
        Debug.Print &B100
        Debug.Print &B111
    End Sub
```

Output:
```
 1 
 2 
 4 
 7 
```

**Reply by VBFNiya** [reply]
Ah ok. Just wanted to make sure. This is something that I wished for for decades throughout my time with BASIC. The necessity comes up a lot when I'm trying to work out bitwise calculations and I need to know what bits are being looked at.
